### PR TITLE
fix: Possible multithreading issue when two scores may come in parallel they may not safely fill pointer values

### PR DESCRIPTION
**Pull request by JoanFM**
fix: Possible multithreading issue when two scores may come in parallel they may not safely fill pointer values

Two threads can fill up scoredDataOutPtr in parallel and mix up their values

### CODE REVIEW

**Comment by imatiach-msft** [comment]
@JoanFM it looks like this needs to be merged with latest?  I think I also see some code that has been merged already in the other PR.

**Comment by JoanFM** [comment]
> @JoanFM it looks like this needs to be merged with latest? I think I also see some code that has been merged already in the other PR.

Yes, I am merging it

File: src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMBooster.scala [hunk 1]
```diff
@@ -9,97 +9,143 @@ import org.apache.spark.ml.linalg.{DenseVector, SparseVector, Vector}
...
+  lazy val data64bitType: Int = lightgbmlibConstants.C_API_DTYPE_FLOAT64
 
   private def getNumClasses: Int = {
     val numClassesOut = lightgbmlib.new_intp()
     LightGBMUtils.validate(
       lightgbmlib.LGBM_BoosterGetNumClasses(boosterPtr, numClassesOut),
       "Booster NumClasses")
-    lightgbmlib.intp_value(numClassesOut)
+    val out = lightgbmlib.intp_value(numClassesOut)
+    lightgbmlib.delete_intp(numClassesOut)
```
Comment (imatiach-msft) [inline src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMBooster.scala]: nice catch to delete the pointer here!

**Pull request merged**